K8s docs v2 (#643)
ngrok/ngrok-docs: docs/k8s/getting-started-gwapi.mdx

---
title: Getting Started - Kubernetes Gateway API
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

import K8sIngressInstallExample from "/examples/k8s/install-ingress.mdx";
import K8sGatewayAPIInstallExample from "/examples/k8s/install-gateway-api.mdx";

import K8sIngressModifyHeadersExample from "/examples/k8s/http-request-headers.mdx";
import K8sGatewayModifyHeadersExample from "/examples/k8s/http-request-headers-gateway.mdx";

import K8sIngressRedirectExample from "/examples/k8s/http-request-redirect.mdx";
import K8sGatewayRedirectExample from "/examples/k8s/http-request-redirect-gateway.mdx";

import StaticDomainKubernetesExample from "/examples/k8s/http-static-domain.mdx";
import StaticDomainKubernetesGatewayExample from "/examples/k8s/http-static-domain-gateway.mdx";

import StaticDomainKubernetesPathExample from "/examples/k8s/http-static-domain-path.mdx";
import StaticDomainKubernetesGatewayPathExample from "/examples/k8s/http-static-domain-gateway-path.mdx";

# Kubernetes Gateway API

## Overview

> The Kubernetes Operator allows access to k8s Services using the new [Gateway API](https://gateway-api.sigs.k8s.io/).

Reviewer: Let's use Kubernetes everywhere instead of `k8s` and `K8s` unless it's in a URL or copied / quoted text.

Reviewer 2: ```suggestion
The Kubernetes Operator allows access to Kubernetes Services using the new [Gateway API](https://gateway-api.sigs.k8s.io/).
```

> This implementation includes CRDS for a Gateway as well as HTTPRoute.

Reviewer 2: ```suggestion
This implementation includes CRDS for a Gateway as well as `HTTPRoute`.
```

> Additional routes, such as TLSRoute and TCPRoute will be added at a later date when they move from experimental to stable.

Reviewer 2: ```suggestion
Additional routes, such as `TLSRoute` and `TCPRoute` will be added at a later date when they move from experimental to stable.
```

## Install

Installing the ngrok Kubernetes Operator is easy using Helm.

<K8sGatewayAPIInstallExample />

> <br />
> 

Reviewer: ```suggestion
```

## Create an edge with a domain

> This would create an Edge on with the Domain `example.ngrok.app` that routes traffic to the Kubernetes service `example-service` running on port `80`.

Reviewer: ```suggestion
This would create an Edge with the Domain `example.ngrok.app` that routes traffic to the Kubernetes service `example-service` running on port `80`.
```

<StaticDomainKubernetesGatewayExample />

## Route Traffic based on a path

> This would create an Edge on with the Domain `example.ngrok.app` that routes traffic from `example.ngrok.app/foo` to the Kubernetes service `example-service` running on port `80`.

Reviewer: ```suggestion
This would create an Edge with the Domain `example.ngrok.app` that routes traffic from `example.ngrok.app/foo` to the Kubernetes service `example-service` running on port `80`.
```

<StaticDomainKubernetesGatewayPathExample />

## Modify Headers

> This would modify the headers by removing the `x-request-id` header, adding an additional one and setting another header to a specific value.

Reviewer: ```suggestion
This would modify request headers by removing the `x-request-id` header, adding an additional one and setting another header to a specific value.
```

<K8sGatewayModifyHeadersExample />

## Redirect Traffic

This would redirect a request from `"example.ngrok.app"` to `"example.ngrok.app/paprika"`

<K8sGatewayRedirectExample />

## Pricing

The ngrok Kubernetes Operator is available to all ngrok users at no

> additional charge. You only incur costs if the resources provisioned by your usage of the Gateway API, such as endpoints or traffic policies, incur a cost. More information can be found on the [pricing page](https://ngrok.com/pricing)

Reviewer 2: ```suggestion
additional charge. You only incur costs if the resources provisioned by your usage of the Kubernetes Operator, such as endpoints or traffic policies, incur a cost. More information can be found on the [pricing page](https://ngrok.com/pricing)
```

Author: This page is specifically targeted at the Gateway API, but I think Operator does make sense in this case.

> additional charge. You only incur costs if the resources provisioned by your usage of the Gateway API, such as endpoints or traffic policies, incur a cost. More information canb be found on the [pricing page](https://ngrok.com/pricing)

Reviewer: ```suggestion
additional charge. You only incur costs if the resources provisioned by your usage of the Gateway API, such as endpoints or traffic policies, incur a cost. More information can be found on the [pricing page](https://ngrok.com/pricing)
```

---
title: Getting Started - Kubernetes Gateway API
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

import K8sIngressInstallExample from "/examples/k8s/install-ingress.mdx";
import K8sGatewayAPIInstallExample from "/examples/k8s/install-gateway-api.mdx";

import K8sIngressModifyHeadersExample from "/examples/k8s/http-request-headers.mdx";
import K8sGatewayModifyHeadersExample from "/examples/k8s/http-request-headers-gateway.mdx";

import K8sIngressRedirectExample from "/examples/k8s/http-request-redirect.mdx";
import K8sGatewayRedirectExample from "/examples/k8s/http-request-redirect-gateway.mdx";

import StaticDomainKubernetesExample from "/examples/k8s/http-static-domain.mdx";
import StaticDomainKubernetesGatewayExample from "/examples/k8s/http-static-domain-gateway.mdx";

import StaticDomainKubernetesPathExample from "/examples/k8s/http-static-domain-path.mdx";
import StaticDomainKubernetesGatewayPathExample from "/examples/k8s/http-static-domain-gateway-path.mdx";

> # Kubernetes Ingress Controller

Reviewer 2: This is titles the k8s ingress controller, but you are talking about the gateway API in the next section.

## Overview

The Kubernetes Operator allows access to k8s Services using the new [Gateway API](https://gateway-api.sigs.k8s.io/).

> This implementation includes CRDS for a Gateway as well at HTTPRoute.

Reviewer: ```suggestion
This implementation includes CRDS for a Gateway as well as HTTPRoute.
```

> Additional routes, such as TLSRoute and TCPRoute will be added at a later date when they move from exerimental to stable.

Reviewer: ```suggestion
Additional routes, such as TLSRoute and TCPRoute will be added at a later date when they move from experimental to stable.
```

## Install

> Installing the ngrok Operator is easy using Helm.

Reviewer 2: ```suggestion
Installing the ngrok Kubernetes Operator is easy using Helm.
```

<K8sGatewayAPIInstallExample />

<br />

## Create an edge with a domain

> This would create an Edge on with the Domain `example.ngrok.app` that routes traffic to the Kubernetes service `example-service` running on port 80.

Reviewer: ```suggestion
This would create an Edge on with the Domain `example.ngrok.app` that routes traffic to the Kubernetes service `example-service` running on port `80`.
```

<StaticDomainKubernetesGatewayExample />

## Route Traffic based on a path

> This would create an Edge on with the Domain `example.ngrok.app` that routes traffic from example.ngrok.app/foo to the Kubernetes service `example-service` running on port 80.

Reviewer: ```suggestion
This would create an Edge on with the Domain `example.ngrok.app` that routes traffic from `example.ngrok.app/foo` to the Kubernetes service `example-service` running on port `80`.
```

<StaticDomainKubernetesGatewayPathExample />

## Modify Headers

> This would modify the headers

Reviewer: Incomplete sentence

<K8sGatewayModifyHeadersExample />

## Redirect Traffic

> This would redirect a request to another URL

Reviewer: Perhaps needs a colon?

<K8sGatewayRedirectExample />

## Pricing

> The ngrok Kubernetes Gateway API is available to all ngrok users at no

Reviewer 2: the ngrok k8s operator is available to all users, which allows users to work with the k8s gateway api.

Author: I modified this to clarify that the operator is free, but the resources used for the Gateway API are what includ the cost.

additional charge. You only incur costs if the resources provisioned by the

> controller incur a cost.

Reviewer 2: maybe add a link to the pricing page?

Reviewer 2: also, what are some of the limitations of using the k8s operator on a free plan?

Author: Added a link. I think there aren't any explicit restrictions but endpoints and traffic policies seem to be the items that would be charged for.
